### PR TITLE
pytorch bindings

### PR DESCRIPTION
**Pull request by Emvlt**
Bug appears only in 3D

### CODE REVIEW

**Comment by AnderBiguri** [comment]
Thanks @Emvlt , I'll have a look. AFAIK: 2D works, 3D crashes.

**Comment by Emvlt** [comment]
So, the crash only happens when forward is called a second time. I can compute a first forward pass exactly the way it should be, but the second call always crashes in 3D. The problematic line is 
` x = x.detach().cpu().numpy()`

**Comment by AnderBiguri** [comment]
I am a bit confused on the different way you tread the 3D version and the 2D version, can you explain further?

In TIGRE, there is no 2D operators, there are only 3D operators that can take 3 rd dimension == 1, so a priori, there is no need to change anything in the way you call  Ax(). 

Would it not make it easier to treat these the same way? i.e. always assume you get a 3D tensor (+batch) as input. This makes it 4D for the THREE_D image case right? why do you have 5D sometimes in the THREE_D case?

**Comment by Emvlt** [comment]
This pull request accounts for:

- Forward and Backward Operators as torch.nn.Modules
- Demo 23: CT reconstruction by iterative data discrepancy minimisation (there is a TODO for @AnderBiguri in this file + there is a known bug for 3D)
- Demo 24: LPD training on MNIST data
